src/overlays/actors/ovl_En_Kusa/z_en_kusa.c

```diff
@@ -20,59 +50,436 @@ const ActorInit En_Kusa_InitVars = {
+            this->collider.base.maskA |= 8;
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kusa/func_80A9B174.s")
+        if (this->actor.xzDistFromLink < 600.0f) {
+            Collider_CylinderUpdate(&this->actor, &this->collider);
+            CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kusa/func_80A9B1FC.s")
+            if (this->actor.xzDistFromLink < 400.0f) {
+                CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+                if (this->actor.xzDistFromLink < 100.0f) {
+                    func_8002F580(&this->actor, globalCtx);
+                }
+            }
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kusa/func_80A9B21C.s")
+void EnKusa_SetupLiftedUp(EnKusa* this) {
+    EnKusa_SetupAction(this, EnKusa_LiftedUp);
+    this->actor.room = -1;
+    this->actor.flags |= 0x10;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kusa/func_80A9B574.s")
+void EnKusa_LiftedUp(EnKusa* this, GlobalContext* globalCtx) {
+    if (Actor_HasNoParent(&this->actor, globalCtx)) {
+        this->actor.room = globalCtx->roomCtx.curRoom.num;
+        func_80A9BBB0(this);
+        this->actor.velocity.x = this->actor.speedXZ * Math_Sins(this->actor.posRot.rot.y);
+        this->actor.velocity.z = this->actor.speedXZ * Math_Coss(this->actor.posRot.rot.y);
+        this->actor.colChkInfo.mass = 240;
+        this->actor.gravity = -0.1f;
+        EnKusa_Fall(this);
+        func_80A9B174(&this->actor.velocity, 0.005f);
+        func_8002D7EC(&this->actor);
+        func_8002E4B4(globalCtx, &this->actor, 7.5f, 35.0f, 0.0f, 0xC5);
+        this->actor.gravity = -3.2f;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kusa/func_80A9B630.s")
+void func_80A9BBB0(EnKusa* this) {
+    EnKusa_SetupAction(this, func_80A9BC1C);
+    D_80A9C1D0 = -0xBB8;
+    D_80A9C1D8 = ((Math_Rand_ZeroOne() - 0.5f) * 1600.0f);
+    D_80A9C1D4 = 0;
+    D_80A9C1DC = 0;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kusa/EnKusa_Init.s")
+void func_80A9BC1C(EnKusa* this, GlobalContext* globalCtx) {
+    s32 pad;
+    Vec3f contactPos;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kusa/EnKusa_Destroy.s")
+    if (this->actor.bgCheckFlags & 11) {
+        if (!(this->actor.bgCheckFlags & 32)) {
+            Audio_PlaySoundAtPosition(globalCtx, &this->actor.posRot.pos, 20, NA_SE_EV_PLANT_BROKEN);
+        }
+        EnKusa_SpawnFragments(this, globalCtx);
+        EnKusa_DropCollectible(this, globalCtx);
+        switch (this->actor.params & 3) {
+            case 0:
+            case 2:
+                Actor_Kill(&this->actor);
+                return;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kusa/func_80A9B7EC.s")
+            case 1:
+                func_80A9BF3C(this);
```

Comment:
Last case doesn't need a break, but it's always good to have one if possible, so it's a bit easier to add more cases when modding, among other things.
```suggestion
                func_80A9BF3C(this);
                break;
```